### ISSUE TITLE
--path feature for install cli not working as expected

### ISSUE THREAD

**Issue by naina-verma** [projectatomic/vagrant-service-manager#376]
--path feature for install cli not working as expected
Find issue details : https://gist.github.com/naina-verma/e03bdfb7bc60adaa22b2dbb75e48f604
tested on latest vsm 
`vagrant-registration (1.2.3)
  Version Constraint: 1.2.3
vagrant-service-manager (1.3.0.dev)
   Version Constraint: 1.3.0.dev
vagrant-sshfs (1.2.0)`

**Comment by LalatenduMohanty** [comment]
Copying the actuall output from gist @naina-verma mentioned . Check for `The system cannot find the path specified.` in below output

```
cdk@WIN10 ~/naina/unify/pk
$ vagrant.exe service-manager install-cli  docker --cli-version 1.10.3 --path /home/cdk/naina/docker
The system cannot find the path specified.
# Binary now available at /home/cdk/naina/docker
# run binary as:
# docker.exe <command>
export PATH=/home/cdk/naina:$PATH

# run following command to configure your shell:
# eval "$(VAGRANT_NO_COLOR=1 vagrant service-manager install-cli docker | tr -d '\r')"
```

**Comment by hferentschik** [comment]
@budhrg what's the state on this. Is the _--path_ option available or not. It is not documented at least.

**Comment by brgnepal** [comment]
@hferentschik Ya not properly documented. Will work with @Preeticp on it.

**Comment by hferentschik** [comment]
Well, afaict there is also no help in the CLI itself. So it is not just a docs issue for sure. Also, the issue that the help system is only explorable if the VM is running still needs addressing.

**Comment by brgnepal** [comment]
Related to #326

**Comment by brgnepal** [comment]
Fixed in #391

**Comment by brgnepal** [comment]
@hferentschik I was wrong here about the --path option value

The value can be any path which doesn't need permission.
like  `$HOME/unknown/dir/docker`

Do you think the install-cli command should create those intermediate directories?

In case the path need permission, the command will fail with system permission error.

**Comment by hferentschik** [comment]
> The value can be any path which doesn't need permission like $HOME/unknown/dir/docker

Sure. That was my expectation anyways. Provided the path exists and is writable there is no reason why we could not copy the binary there.

> Do you think the install-cli command should create those intermediate directories?

Not sure. I guess both works. Thinking about other tools, I think, however, that is is more common practice to expect the directory to exist. It also avoids the case that we create a recursive directory structure due to some simple typo in the path. WDYT on this one?

> In case the path need permission, the command will fail with system permission error.

So assuming we require the path to exist, our code would verify it exists and is writable. If not an error would be logged and the command aborts. If we allow the path to not exist, we would create it recursively. 

On a side note, do we properly set the executable bit on the binary? It would imo make sense to verify that the file is executable after copying it into the specified location. This is regardless of _--path_ is used or not.

**Comment by brgnepal** [comment]
> On a side note, do we properly set the executable bit on the binary

Yes, we are making it executable. I think this [test definition](https://github.com/projectatomic/vagrant-service-manager/blob/master/features/support/env.rb#L153) is checking it so no need to worry :smile:

**Comment by hferentschik** [comment]
> Yes, we are making it executable. I think this test definition is checking it so no need to worry 😄

👍

**Comment by hferentschik** [comment]
Resolved via pull request #416
